src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala

```diff
@@ -73,21 +73,25 @@ object BasicOpcDiff extends OpcDiffer {
 
   def convertion(x: DataEntry, t: DataEntry): Either[ValidationError, DataEntry] =
     t.dataType match {
-      case DataTypeObj => x.dataType match {
-        case Int32 | Amount | Timestamp | BigInteger => {
+      case DataTypeObj =>(x.dataType match {
+        case Int32 | Amount | Timestamp => Right(x.data) 
+        case BigInteger => Right(x.data.drop(2))
+        case _ => Left(ContractUnsupportedOPC)
+      }) flatMap { bytes => {
           val to: DataTypeVal[_] = DataTypeObj.deserializer(t.data)
           to match {
-            case Int32      => formatResB[Int]   (x.data, Int32)
-            case Amount     => formatResB[Long]  (x.data, Amount)
-            case Timestamp  => formatResB[Long]  (x.data, Timestamp)
-            case BigInteger => formatResB[BigInt](x.data, BigInteger)
+            case Int32      => formatResB[Int]   (addLeadingZeros(bytes, 4), Int32)
```

Comment:
In this logic, if we convert from Long to Int, it needs to reduce the leading zeros~